### ISSUE TITLE
ArgumentOutOfRangeException with multiple TagWith calls

### ISSUE THREAD

**Issue by henrym** [VariableNotFound/ef6-tagwith#1]
```
var tag1 = nameof(QueryOnlyMultipleTagWiths);
var tag2 = "Second tag";

var query = _ctx.Friends
                .TagWith(tag1)
                .TagWith(tag2)
                .ToList();
```
Specifically when the second predicate is at the end of the generated SQL. The `CmpStr` method has a bug when `startIndex + compare.Length` is longer that `sql` and it returns true for `endsWithAnd`.

The above test case also exhibits another bug where the second predicate to be removed from the SQL is wrapped with parens: `WHERE (N'!__tag!' == N'tag text')` but the removal code doesn't take this into account and the resulting SQL contains `W)` in it.

**Comment by jmaguilar** [comment]
Good catch, thank you for the pull request!

**Comment by henrym** [comment]
Thanks José!

**Comment by henrym** [comment]
Hi José, it seems the 1.0.3 NuGet package still has the old version of the code. If it's not too much trouble could you please make a new release with this bug fixed?

Thank you

**Comment by jmaguilar** [comment]
Oops, sorry! I've just published the 1.0.4 package.

Please let me know if you find new issues.

Thanks for reporting!
